odoo/models.py

```diff
@@ -6150,22 +6150,35 @@ def _get_placeholder_filename(self, field=None):
         return 'placeholder.png'
 
     def _populate_factories(self):
-        """ Return a list of pairs ``(field_name, factory)``, ``factory`` is a function that creates
-        a generator of values (dict of field values).  Its API suggested by::
+        """ Return a list of pairs ``(field_name, factory)``.
 
-            # generator yield values by iterating on base_generator and adding a value
-            # for the given field in the dicts yielded by base_generator
-            generator = factory(base_generator, field_name, model_name)
-            # suggested usage
-            for values in generator:
-                model.create(values)
+        ``factory`` is a generator of values (dict of field values).
+
+        Factory skeleton::
+
+            def generator(iterator, field_name, model_name):
+                for counter, values in enumerate(iterator):
+                    # values.update(dict())
+                    yield values
+
+        See :mod:`odoo.tools.populate` for population tools and applications.
+        This method must be overridden on a model to support database population.
+
+        :returns: list of pairs(field_name, factory) where `factory` is a generator function.
+        :rtype: list(tuple(str, generator))
         """
         return []
 
     @property
     def _populate_sizes(self):
         """ Return a dict mapping symbolic sizes (``'small'``, ``'medium'``, ``'large'``) to integers,
-        giving the minimal number of records that method ``_populate`` should create.
+        giving the minimal number of records that :meth:`_populate` should create.
```

Comment:
Why's this a property but the custom model above used a class attribute?

Reply:
Both works, and it's easier/faster to define a class attribute :laughing: (In the models currently supporting the population, it's already done like this, but we could probably replace the property by a class attribute in models.py).